django/db/backends/base/base.py

```diff
@@ -39,6 +39,34 @@ def timezone_constructor(tzname):
     return zoneinfo.ZoneInfo(tzname)
 
 
+@contextmanager
+def debug_transaction_end(connection, transaction_end):
+    start = time.monotonic()
+    try:
+        yield
+    finally:
+        if connection.queries_logged:
+            stop = time.monotonic()
+            duration = stop - start
+            connection.queries_log.append(
+                {
+                    "sql": "%s" % transaction_end,
+                    "time": "%.3f" % duration,
+                }
+            )
+            logger.debug(
+                "(%.3f) %s; alias=%s",
+                duration,
+                transaction_end,
+                connection.alias,
+                extra={
+                    "duration": duration,
+                    "sql": transaction_end,
+                    "alias": connection.alias,
+                },
+            )
```

Comment:
`CursorDebugWrapper.debug_sql` uses `django.db.backends`. I think we should use the same logger here, update [docs](https://docs.djangoproject.com/en/dev/ref/logging/#django-db-backends):

_"This logging does not include framework-level initialization (e.g. SET TIMEZONE) or transaction management queries (e.g. BEGIN, **COMMIT**, and **ROLLBACK**)."_

and add a `versionchanged` annotation.

Reply:
cool, thanks for taking a look, I will follow-up within a few days.

Reply:
Moved the function to utils.py which allowed to use the proper logger without defining two in base.
Fixed the docs.

Reply:
oh, BEGIN is too specific for Postgres.  Removing that assertion.

Reply:
`BEGIN` is logged only on SQLite as a workaround to start a transaction explicitly in autocommit mode (see [comment](https://code.djangoproject.com/ticket/31090#comment:4)). I think we should use `debug_transaction_end` to wrap 

https://github.com/django/django/blob/c765b62e3258de4dce9935ab7aed430346dfbc10/django/db/backends/base/base.py#L489-L492

when `autocommit` is `False`.

Reply:
As I understand it you mean wrap the _start_transaction_under_autocommit call?  That is done.

Reply:
@felixxm thanks for the feedback, I think I addressed everything, so when you get a chance perhaps you can take another look.

Reply:
`BEGIN` is called explicitly in `_start_transaction_under_autocommit()` so there is no need to log it twice. We should wrap `self._set_autocommit()` when `autocommit` is False as the usual way to start a transaction is to turn `autocommit` off. I pushed edits.